src/xenia/app/emulator_window.cc

```diff
@@ -928,6 +936,41 @@ void EmulatorWindow::ToggleFullscreen() {
   SetFullscreen(!window_->IsFullscreen());
 }
 
+void EmulatorWindow::TakeScreenshot() {
+  xe::ui::RawImage myimage;
+  if (emulator()->is_title_open()) {  // Make sure we have a game open first
+    GetGraphicsSystemPresenter()->CaptureGuestOutput(myimage);
```

Comment:
Note that the guest output image, depending on the game, may have a different aspect ratio than the display. Halo 3 provides a 1152x640 image (1.8), for instance, while 16:9 is 1.777…

Reply:
I think for purpose of debugging, this is a perfectly fine limitation. The intent is to capture only what's actually being rendered by the emulator, not the entire display. Anything beyond the bounds of the guest output image would be a black letterbox anyway.